Simplified/Network/NYPLRemoteViewController.m

```diff
@@ -128,54 +130,38 @@ - (void)load
   self.dataTask = [NYPLNetworkExecutor.shared addBearerAndExecute:request
                            completion:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
 
+    NSHTTPURLResponse *httpResponse = nil;
+    if ([response isKindOfClass: [NSHTTPURLResponse class]]) {
+      httpResponse = (NSHTTPURLResponse *) response;
+    }
     dispatch_async(dispatch_get_main_queue(), ^{
       [self.activityIndicatorView stopAnimating];
       self.activityIndicatorLabel.hidden = YES;
       NSURLSessionDataTask *dataTaskCopy = self.dataTask;
       self.dataTask = nil;
 
-      __auto_type user = [NYPLUserAccount sharedAccount];
-
-      __block NYPLReauthenticator *reauthenticator = [[NYPLReauthenticator alloc] init];
-      BOOL authenticating = [reauthenticator authenticateUser:user
-                                          ifNeededForResponse:response
-                                                 responseData:data
-                                                responseError:error
-                                      authenticationPreflight:^{ self.reloadView.hidden = NO; }
-                                     authenticationCompletion:^{
-        // make sure to retain the reauthenticator until end of auth flow and
-        // then break any possible retain cycle
-        reauthenticator = nil;
-        NYPLLOG(@"Re-loading from RemoteVC because authentication expired");
-        [self load];
```

Comment:
this was an async infinite loop. Now we just fail, show the retry button, and record the need to reauthenticate in case we failed for that reason. Note that `needsReauthentication` is only accessed from main thread.